pkg/jsonnet/lint.go

```diff
@@ -37,42 +44,21 @@ func Lint(fds []string, opts *LintOpts) error {
 	}
 
 	type result struct {
-		failed bool
-		output string
+		success bool
+		output  string
 	}
 	fileCh := make(chan string, len(paths))
 	resultCh := make(chan result, len(paths))
 	lintWorker := func(fileCh <-chan string, resultCh chan result) {
 		for file := range fileCh {
-			buf := &bytes.Buffer{}
-			var err error
-			file, err = filepath.Abs(file)
-			if err != nil {
-				fmt.Fprintf(buf, "got an error getting the absolute path for %s: %v\n\n", file, err)
-				resultCh <- result{failed: true, output: buf.String()}
-				continue
-			}
-
-			log.Debug().Str("file", file).Msg("linting file")
-			startTime := time.Now()
-
-			vm := MakeVM(Opts{})
-			jpaths, _, _, err := jpath.Resolve(file, true)
-			if err != nil {
-				fmt.Fprintf(buf, "got an error getting JPATH for %s: %v\n\n", file, err)
-				resultCh <- result{failed: true, output: buf.String()}
-				continue
-			}
-
-			vm.Importer(NewExtendedImporter(jpaths))
-
-			content, _ := os.ReadFile(file)
-			failed := linter.LintSnippet(vm, buf, []linter.Snippet{{FileName: file, Code: string(content)}})
-			resultCh <- result{failed: failed, output: buf.String()}
-			log.Debug().Str("file", file).Dur("duration_ms", time.Since(startTime)).Msg("linted file")
+			buf, success := lintWithRecover(file)
+			resultCh <- result{success: success, output: buf.String()}
 		}
 	}
 
+	if opts.Parallelism <= 0 {
+		opts.Parallelism = 4
+	}
```

Comment:
Considering that we are using the same default as when one is not specified, should we check this in `cmd/tk/lint.go`? Or alternatively check at the beginning of the function an error?

Reply:
True, it has a default already in `cmd`. I converted it to an error instead

Reply:
👍